hicommonwealth/commonwealth
# 4682 - Create Table Component

The work here completes issue #4682. #4682 specifies the creation of a table component that can be used in the (not-yet-implemented) communities directory page. The table must provide the ability to set columns as sortable or not sortable. The table cells can display the following types of data: numbers, text, avatars alongside text, tags, and buttons. The table is initially sorted by its first column.

Note that, though our first use of this component will be the communities page, it can be used on any page, so long as data is passed to it in the manner specified in the comments in `/component_kit/cw_table.tsx`. 

## Link to Issue
Closes: #4682 

## Description of Changes
Creates a table with:
- the ability to configure a column as sortable
- the ability to configure a column as numerical (the text in numerical cells is displayed differently)
- the ability to display different types of data; specifically, text, numbers, avatars, tags, and buttons

## Test Plan
- navigate to [commonwealth demo components showcase](https://commonwealth-demo.herokuapp.com/components). The table component can be found at the bottom of the page
- Note that the table is initially sorted by the first column
- click on the sortable columns to check that the sorting occurs as expected
- click on the non-sortable columns (the ones without arrows) and check that nothing happens
